app/file.go

```diff
@@ -76,12 +76,36 @@ func (a *App) FileBackend() (filesstore.FileBackend, *model.AppError) {
 	return a.Srv().FileBackend()
 }
 
+func (a *App) CheckMandatoryS3Fields(settings *model.FileSettings) *model.AppError {
+	err := filesstore.CheckMandatoryS3Fields(settings)
+	if err != nil {
+		return model.NewAppError("CheckMandatoryS3Fields", "api.admin.test_s3.missing_s3_bucket", nil, "", http.StatusBadRequest)
```

Comment:
nit: I understand that we are currently returning error only if s3 bucket is not exsists, but this error seems a little bit misleading to me. We should return that detail from CheckMandatoryS3Fields, and say something more generic here.

Reply:
I think you are right, but I don't want to interfere too much with the generated errors here, because I think that are user facing.